pallets/hsm/src/lib.rs

```diff
@@ -0,0 +1,1350 @@
+	fn burn_hollar(amount: Balance) -> DispatchResult {
+		let contract = T::GhoContractAddress::contract_address(T::HollarId::get())
+			.ok_or(Error::<T>::HollarContractAddressNotFound)?;
+		let pallet_address = T::EvmAccounts::evm_address(&Self::account_id());
+
+		// Create the context for the EVM call
+		let context = CallContext::new_call(contract, pallet_address);
+
+		// Encode the burn function call with amount
+		let mut data = Into::<u32>::into(ERC20Function::Burn).to_be_bytes().to_vec();
+		data.extend_from_slice(H256::from_uint(&U256::from(amount)).as_bytes());
+
+		// Execute the EVM call
+		let (exit_reason, value) = T::Evm::call(context, data, U256::zero(), T::GasLimit::get());
+
+		// Check if the call was successful
+		if exit_reason != ExitReason::Succeed(ExitSucceed::Stopped) {
+			log::error!(target: "hsm", "Burn Hollar EVM execution failed. Reason: {:?}", value);
+			return Err(Error::<T>::InvalidEVMInteraction.into());
+		}
+
+		Ok(())
+	}
+
+	/// Simulates selling an asset with exact input in a StableSwap pool
+	///
+	/// Calculates how much of asset_out would be received by selling a specific amount of asset_in.
+	/// Uses the StableSwap pool mechanism with the provided pool state snapshot.
+	fn simulate_out_given_in(
+		pool_id: T::AssetId,
+		asset_in: T::AssetId,
+		asset_out: T::AssetId,
+		amount_in: Balance,
+		min_amount_out: Balance,
+		pool_state: &PoolSnapshot<T::AssetId>,
+	) -> Result<Balance, DispatchError> {
+		let (amount_out, _) = pallet_stableswap::Pallet::<T>::simulate_sell(
+			pool_id,
+			asset_in,
+			asset_out,
+			amount_in,
+			min_amount_out,
+			Some(pool_state.clone()),
+		)?;
+		Ok(amount_out)
+	}
+
+	/// Simulates buying an asset with exact output in a StableSwap pool
+	///
+	/// Calculates how much of asset_in would be required to buy a specific amount of asset_out.
+	/// Uses the StableSwap pool mechanism with the provided pool state snapshot.
+	fn simulate_in_given_out(
+		pool_id: T::AssetId,
+		asset_in: T::AssetId,
+		asset_out: T::AssetId,
+		amount_out: Balance,
+		max_amount_in: Balance,
+		pool_state: &PoolSnapshot<T::AssetId>,
+	) -> Result<Balance, DispatchError> {
+		let (amount_in, _) = pallet_stableswap::Pallet::<T>::simulate_buy(
+			pool_id,
+			asset_in,
+			asset_out,
+			amount_out,
+			max_amount_in,
+			Some(pool_state.clone()),
+		)?;
+		Ok(amount_in)
+	}
+
+	/// Process arbitrage opportunities for all collateral assets
+	///
+	/// This function:
+	/// 1. Acquires a lock to prevent concurrent execution
+	/// 2. Checks each collateral asset for arbitrage opportunities
+	/// 3. Submits unsigned transactions to execute profitable arbitrages
+	///
+	/// This is called from the offchain worker to maintain the Hollar peg.
+	fn process_arbitrage_opportunities(block_number: BlockNumberFor<T>) -> Result<(), DispatchError> {
+		let lock_expiration = Duration::from_millis(LOCK_TIMEOUT);
+		let mut lock = StorageLock::<'_, Time>::with_deadline(OFFCHAIN_WORKER_LOCK, lock_expiration);
```

Comment:
Just a thought , wouldn't normal lock(without deadline) be better? Theoretically this lock can expire during inner loop execution. I'm not how big of an issue it would be.